chirho/robust/internals/nmc.py

```diff
@@ -205,7 +162,11 @@ def forward(
 
         # move data plate dimension to the left
         for name in reversed(plate_name_to_dim.keys()):
-            log_weights = bind_leftmost_dim(log_weights, name)
+            log_weights = torch.transpose(
```

Comment:
This is the only non-renaming change in this PR. It is equivalent to the previous version with the definition of `bind_leftmost_dim` inlined for this special case.

Reply:
I have a slight preference for keeping it as `bind_leftmost_dim`, as that seems a bit more transparent about what the code is doing.

If we do decide to remove it here is `bind_leftmost_dim` used anywhere else anymore?

Reply:
I inlined `bind_leftmost_dim` here because it is now in `observational.internals` and shouldn't be used outside `observational`. I could have moved `BatchedNMCLogMarginalLikelihood`, but I would like to keep that as an internal implementation detail of `chirho.robust` for now.

`bind_leftmost_dim` is still used inside `chirho.observational.handlers.predictive.PredictiveFunctional`, which was moved in this PR.